numpy/lib/function_base.py

```diff
@@ -1595,22 +1606,58 @@ def trim_zeros(filt, trim='fb'):
     [1, 2]
 
     """
-    first = 0
-    trim = trim.upper()
-    if 'F' in trim:
-        for i in filt:
-            if i != 0.:
-                break
-            else:
-                first = first + 1
-    last = len(filt)
-    if 'B' in trim:
-        for i in filt[::-1]:
-            if i != 0.:
-                break
+    trim = trim.lower()
+
+    absolutes = np.abs(filt)
+    nonzero = np.nonzero(absolutes)
+    lengths = np.zeros((absolutes.ndim, 2), dtype=np.intp)
+
+    if axis is None:
+        # Apply iteratively to all axes
+        axis = range(absolutes.ndim)
+    # Normalize axes to 1D-array
+    axis = np.asarray(axis, dtype=np.intp)
+    if axis.ndim == 0:
+        axis = np.asarray([axis], dtype=np.intp)
```

Comment:
I think you want `normalize_axis_tuple` here, which also removes the need to call `normalize_axis_index` in the loop.